.github/workflows/python-security.yml

```diff
@@ -0,0 +1,58 @@
+# Workflow regarding python security scanning.
+name: Python Security
+
+on:
+  push:
+    branches:
+      # Run on our main branch
+      - main
+    paths:
+      - example-python/**
+  pull_request:
+    # Run for any pull requests
+    paths:
+      - example-python/**
+jobs:
+  # https://github.com/marketplace/actions/anchore-container-scan
+  dependency-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4  # Checkout the current branch/merge state
+      - name: Grype Scan
+        uses: anchore/scan-action@v3
+        with:
+          path: example-python
+      - name: Pip Audit
+        uses: pypa/gh-action-pip-audit@v1.1.0
+        with:
+          inputs: example-python
+  code-check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.13"]
+    permissions:
+      # Required for uploading sarif file
+      security-events: write
+    steps:
+      - uses: actions/checkout@v4  # Checkout the current branch/merge state
+      - name: Set up Python ${{ matrix.python-version }}  # Get Python ready to use
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies  # Install bandit
+        working-directory: ./example-python
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install bandit[toml,sarif]
+      - name: Run Bandit
+        working-directory: ./example-python
+        # Run bandit and output to sarif file
+        run: |
+          bandit -r . -c pyproject.toml -f sarif -o bandit.sarif
+      - name: Upload sarif file
```

Comment:
Same question here where is the `bandit.sarif` file ? Probably, I just don't know where to look (I'm more accustomed to Gitlab)

Reply:
@leobr29 So the bandit report can be found here: https://github.com/British-Oceanographic-Data-Centre/amrit-repos/security/code-scanning?query=pr%3A11+is%3Aopen

it basically creates alerts based on the content of the uploaded sarif file. So if there are no errors, nothing is created. So you can see there are 2 closed alerts (as I have fixed these ones after testing).